### PR TITLE
feat: add data source google_compute_addresses

### PR DESCRIPTION
**Pull request by xabufr**
Fixes hashicorp/terraform-provider-google#10212
Implement a data source to list google_compute_addresses using filters

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-datasource
`google_compute_addresses`
```

### CODE REVIEW

**Comment by slevenick** [comment]
Looks like this fails to compile due to:

```
mmv1/provider/file_template.rb:73:in `generate': google/data_source_google_compute_addresses_test.go missing autogen (RuntimeError)
	from mmv1/provider/core.rb:189:in `block (2 levels) in compile_file_list'
```

**Comment by xabufr** [comment]
Should be better now, thanks!

**Comment by xabufr** [comment]
Hello @slevenick, any update on this PR?

File: mmv1/third_party/terraform/data_sources/data_source_google_compute_addresses.go.erb [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+						"region": {
+							Type:     schema.TypeString,
+							Computed: true,
+							Optional: true,
+						},
+						"self_link": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+<% unless version == 'ga' -%>
```
Comment (slevenick) [inline mmv1/third_party/terraform/data_sources/data_source_google_compute_addresses.go.erb:60]: Is labels beta-only? The tests include labels for addresses, will this cause problems?

Comment (xabufr) [reply mmv1/third_party/terraform/data_sources/data_source_google_compute_addresses.go.erb:60]: Labels for compute addresses are still in beta yes, yes I forgot to condition labels in tests resource, I'll fix thanks!

**Review by slevenick — APPROVED** [review]

**Pull request merged**